From 71e7b0e0c0ccb927af63c2a881958201aebe27de Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 6 Aug 2026 10:06:17 +0200
Subject: [PATCH] decoder: return early when the object type can't be supported

No need to allocate a filter bank that will not be used (and may leak).
---
 libfaad/decoder.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libfaad/decoder.c b/libfaad/decoder.c
index de3b169..cd7ad10 100644
--- a/libfaad/decoder.c
+++ b/libfaad/decoder.c
@@ -372,6 +372,9 @@ long NeAACDecInit(NeAACDecHandle hpDecoder,
     }
 #endif
 
+    if (can_decode_ot(hDecoder->object_type) < 0)
+        return -1;
+
     /* must be done before frameLength is divided by 2 for LD */
 #ifdef SSR_DEC
     if (hDecoder->object_type == SSR)
@@ -385,9 +388,6 @@ long NeAACDecInit(NeAACDecHandle hpDecoder,
         hDecoder->frameLength >>= 1;
 #endif
 
-    if (can_decode_ot(hDecoder->object_type) < 0)
-        return -1;
-
     return bits;
 }
 
-- 
2.52.0.windows.1

